src/executor/stack/memory.rs

```diff
@@ -383,6 +384,19 @@ impl<'config> MemoryStackSubstate<'config> {
 		self.account_mut(address, backend).code = Some(code);
 	}
 
+	pub fn set_delegation<B: Backend>(
+		&mut self,
+		authority: H160,
+		delegation: Delegation,
+		backend: &B,
+	) {
+		self.account_mut(authority, backend).code = Some(delegation.to_bytes());
+	}
+
+	pub fn remove_delegation<B: Backend>(&mut self, authority: H160, backend: &B) {
+		self.account_mut(authority, backend).code = Some(Vec::new());
```

Comment:
Maybe use `None` here:
```suggestion
		self.account_mut(authority, backend).code = None;
```

Reply:
Actually that doesn't work (the delegation cleanup test would fail). I didn't debug the reason, but found this to be working.